### PR TITLE
Add more unit tests for group by

### PR DESCRIPTION
**Pull request by drcrallen**
In response to https://github.com/druid-io/druid/pull/1532#issuecomment-121786764

### CODE REVIEW

File: processing/src/test/java/io/druid/query/groupby/GroupByQueryRunnerTest.java [hunk 1]
```diff
@@ -338,7 +492,7 @@ public void testGroupByWithSimpleRename()
         .setDimensions(
             Lists.<DimensionSpec>newArrayList(
                 new ExtractionDimensionSpec(
-                    "quality", "alias", new LookupExtractionFn(new MapLookupExtractor(map), false, null, true), null
+                    "quality", "alias", new LookupExtractionFn(new MapLookupExtractor(map), false, "MISSING", true), null
```
Comment (xvrl) [inline processing/src/test/java/io/druid/query/groupby/GroupByQueryRunnerTest.java]: why are we changing this test?

Comment (drcrallen) [inline processing/src/test/java/io/druid/query/groupby/GroupByQueryRunnerTest.java]: This is the test that would have caught the bug for TopN. From a testing perspective, if the second parameter (retain missing values) is `false` both `null` and `"MISSING"` would work here, but if I change the second parameter to `true` (which is what caused the bug to be missed in topN) then only `null` is valid. `"MISSING"` will fail validation.

Essentially, if anyone ever copies around this test for more stuff, it makes it less likely to accidentally flip a parameter.

Comment (xvrl) [inline processing/src/test/java/io/druid/query/groupby/GroupByQueryRunnerTest.java]: It seems a bit confusing to have a value when it's not being used, so someone could be mislead in thinking that there is a reason for that string to exist. Maybe we should have both tests instead, one with true, "MISSING", and the other with false, null.

Comment (drcrallen) [inline processing/src/test/java/io/druid/query/groupby/GroupByQueryRunnerTest.java]: Done

**Pull request merged**